src/analysis/bounded_variation.lean

```diff
@@ -686,60 +688,133 @@ lemma monotone_on.has_locally_bounded_variation_on {f : α → ℝ} {s : set α}
   has_locally_bounded_variation_on f s :=
 λ a b as bs, ((hf.evariation_on_le as bs).trans_lt ennreal.of_real_lt_top).ne
 
+/--
+The **signed** variation of `f` on the interval `Icc a b` intersected with the set `s`,
+squashed to a real (therefore only really meaningful if the variation is finite)
+-/
+noncomputable def variation_on_from_to (f : α → E) (s : set α) (a b : α) : ℝ :=
+if a ≤ b then (evariation_on f (s ∩ Icc a b)).to_real else
+            - (evariation_on f (s ∩ Icc b a)).to_real
+
+namespace variation_on_from_to
+
+variables (f : α → E) (s : set α)
+
+@[protected]
+lemma self (a : α) : variation_on_from_to f s a a = 0 :=
+begin
+  dsimp only [variation_on_from_to],
+  rw [if_pos le_rfl, Icc_self, evariation_on.subsingleton, ennreal.zero_to_real],
+  exact λ x hx y hy, hx.2.trans hy.2.symm,
+end
+
+@[protected]
+lemma nonneg_of_le {a b : α} (h : a ≤ b) : 0 ≤ variation_on_from_to f s a b :=
+by simp only [variation_on_from_to, if_pos h, ennreal.to_real_nonneg]
+
+@[protected]
+lemma eq_neg_swap (a b : α) :
+  variation_on_from_to f s a b = - variation_on_from_to f s b a :=
+begin
+  rcases lt_trichotomy a b with ab|rfl|ba,
+  { simp only [variation_on_from_to, if_pos ab.le, if_neg ab.not_le, neg_neg], },
+  { simp only [self, neg_zero], },
+  { simp only [variation_on_from_to, if_pos ba.le, if_neg ba.not_le, neg_neg], },
+end
+
+@[protected]
+lemma nonpos_of_ge {a b : α} (h : b ≤ a) : variation_on_from_to f s a b ≤ 0 :=
+begin
+  rw eq_neg_swap,
+  exact neg_nonpos_of_nonneg (nonneg_of_le f s h),
+end
+
+@[protected]
+lemma eq_of_le {a b : α} (h : a ≤ b) :
+  variation_on_from_to f s a b = (evariation_on f (s ∩ Icc a b)).to_real := if_pos h
+
+@[protected]
+lemma eq_of_ge {a b : α} (h : b ≤ a) :
+  variation_on_from_to f s a b = - (evariation_on f (s ∩ Icc b a)).to_real :=
+by rw [eq_neg_swap, neg_inj, eq_of_le f s h]
+
+@[protected]
+lemma add {f : α → E} {s : set α} (hf : has_locally_bounded_variation_on f s)
+  {a b c : α} (ha : a ∈ s) (hb : b ∈ s) (hc : c ∈ s) :
+  variation_on_from_to f s a b + variation_on_from_to f s b c = variation_on_from_to f s a c :=
+begin
+  symmetry,
+  refine @additive_of_is_total _ _ _ (≤) _ (λ (x y : s), variation_on_from_to f s x y)
+                               (λ x, true) _ _ ⟨a,ha⟩ ⟨b,hb⟩ ⟨c,hc⟩ trivial trivial trivial,
```

Comment:
You'll need to merge master and fix this; `additive_of_is_total` was changed to take in a set `s`.

Reply:
but why isn't github informing me of this then? Did you change from predicate to set after I merged first?

Reply:
Sorry, I meant predicate. I mean you wouldn't need to pass to the subtype (coerced from the set `s`) and should be able to use `(∈ s)` as the predicate rather than `(λ x, true)`.

Reply:
ah, dumb me, I didn't think of making use of that! thanks